the-collab-lab/tcl-11-smart-shopping-list
# Sd kl duplicate items

## Description

This code modifies the `<AddItem />` component to show error messages for duplicate items. There is data clean up for punctuation and capitalization to ensure correct data comparison. The approach was to use regex to remove punctuation and the `toLowerCase()` built in method for capitalization. This issue was informative related to implementing aspects of data quality. 



## Related Issue

closes issue #6 

## Acceptance Criteria


## Type of Changes

|     | Type                       |
| --- | -------------------------- |
| ✓ | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

## Updates

### Before
![collab_lab_issue_6_before](https://user-images.githubusercontent.com/54756529/90950254-b8398180-e404-11ea-8ac2-0b410e357a3a.gif)


### After
![collab_lab_issue_6_after](https://user-images.githubusercontent.com/54756529/90950273-d7d0aa00-e404-11ea-9d9b-9440757663b8.gif)

## Testing Steps / QA Criteria

- From your terminal, run git checkout `sd-kl-duplicate-items`.
- Once on the `sd-kl-duplicate-items` branch, run npm start to load the browser.
- Visit the `<List/>` component at http://localhost:3000/list.
- See the items in your list.
- Click `Add Item` and add a new item to your list with punctuation or capitalization. 
- Visit the `<List/>` component at http://localhost:3000/list and you should see your item.
- Add the same item to your list again with or without punctuation or capitalization.
- You should receive an error message.
